interview-questions.md

```diff
@@ -7,53 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a block in Ruby?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:A segment of code in ruby is called a block.
 
+Researched answer:Blocks are called closures in other programming languages. There are some important points about Blocks in Ruby: Block can accept arguments and returns a value. ... A block is always invoked with a function or can say passed to a method call. To call a block within a method with a value, yield statement is used.
 
 2. What is a gem?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:Gem's are used to run tests in Ruby.
 
+Researched answer: RubyGems is a package manager for the Ruby programming language that provides a standard format for distributing Ruby programs and libraries, a tool designed to easily manage the installation of gems, and a server for distributing them.
 
 3. What is Ruby on Rails?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:Ruby on Rails is a language that incorporates front and back end.
 
+Researched answer:Ruby on Rails, or Rails, is a server-side web application framework written in Ruby under the MIT License. Rails is a model–view–controller framework, providing default structures for a database, a web service, and web pages.
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:It's a database that you can add information, remove information and edit information and update information.
 
+Researched answer: A relational database is a digital database based on the relational model of data, as proposed by E. F. Codd in 1970. A system used to maintain relational databases is a relational database management system (RDBMS). Many relational database systems have an option of using the SQL (Structured Query Language) for querying and maintaining the database.Databases are widely divided into two major types or categories, namely, Relational or Sequence Databases and Non-relational or Non-sequence databases or No SQL databases. An organization may use them individually or combined, depending on the nature of data and functionality required.
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:I don't know what a primary keys so i can't explain their importance but I'll research so i will know what they are.
 
+Researched answer:In the world of databases, the primary key of a relational table uniquely identifies each record in the table. Databases use keys to compare, sort, and store records, and to create relationships between records. Choosing the primary key in a database is one of the most important steps in the process.
```

Comment:
Nice! It is an important thing to consider. Luckily as Rails developers, we don't have to worry about the primary key. Rails creates them automatically.